src/skillsaw/rule.py

```
@@ -70,15 +76,150 @@ def __str__(self):
+    def target_path(self) -> Path:
+        """File path affected by this fix, used for conflict detection."""
+        ...
+
+    @abstractmethod
+    def apply(self) -> None:
+        """Write the fix to disk through the appropriate node method."""
+        ...
+
+    def diff(self, root: Optional[Path] = None) -> str:
+        """Produce a unified diff string for dry-run display."""
+        ...
```

Ruff (0.15.14)

[warning] `FixOp.diff` is an empty method in an abstract base class, but has no abstract decorator

(B027)